### PR TITLE
Composable image

### PR DESCRIPTION
**Pull request by flovntp**
first draft for composable image on Upsun side (only for now)

- [x] report updates to Platform.sh doc 
- [x] introduction of the composable image
- [x] add Jerome's talk 
- [x] add links in each of the Languages doc pages

Urls: 
Upsun: https://docs.upsun.com.composable-image-o3lpfiy-ucq44jg6ofare.eu-5.platformsh.site/create-apps/app-reference.html
Platform.sh: https://composable-image-o3lpfiy-ucq44jg6ofare.eu-5.platformsh.site/create-apps/app-reference.html

Open question: split each of the top-level YAML keys in dedicated pages as there is a lot of common top level keys between the two (built-in and composable) and merge the 2 (built-in&Comp) with codetabs in a `Runtime` doc page
```
Create Apps
  App Reference
     Primary application properties
     Root directory
     Images
       Built-in image   --> that shows the current App Reference way to define a `type`
       Composable image --> that shows how to define a `stack`
     Resources
     ... 
```
and in each of the other top level key pages, add codetabs with both (Built-in and Composable) config (only differences would be the `type` example in the snippets

And for `type`, `build`, `dependencies`, and `runtime` specific page, add a big warning at the beginning telling the reader that it's not valid if using composable image.

## Why

Closes https://github.com/platformsh/platformsh-docs/issues/3774

### CODE REVIEW

**Comment by lolautruche** [comment]
Here are a few comments:

* "Built-in Image" ➡️ *Single-runtime Image*, as suggested by @AnouckColson and @chadwcarlson 
* "Composable image is a new way to install" ➡️ We should refrain ourselves to use "new", as this doc page is going to last
* We're *not* using NixOS, only *Nix*
* I'm not sure we should describe Nix and NixOS in the doc. I'd better do this in a blog post instead. We may just mention that the Composable Image is built on top of Nix, and that we can install packages from NixPkgs.
* We should mention that the first element of the list will be considered as the "Primary runtime". It's especially important for PHP as PHP-FPM is started automatically _only_ if PHP is specified as the first item.
* `stack` row in the properties table:
  * Type: I'd rephrase to "An array of Nix packages" (+ link to the `stack` doc section)
  * Description: "A list of packages from our collection of supported runtimes and/or from NixPkgs".
* `stack` section:
  * We should probably make a clearer distinction between runtimes and packages
  * "Available tools and their supported version": I think we can remove this part
  * We should have a dedicated section for PHP, presenting all the peculiarities (inner configuration with subsections, how to add an extension...)

File: sites/friday/src/create-apps/app-reference/composable-image.md [hunk 1]
```diff
@@ -0,0 +1,1352 @@
...
+
+If you want to learn more about Nix, please have a look on [Jérôme Vieilledent's presentation](https://www.youtube.com/watch?v=emOt32DVl28) during the Symfony Con Brussels 2023 (our Product Manager of that Composable Image feature).
+
+## Primary application properties
+
+All application configuration takes place in a `{{< vendor/configfile "app" >}}` file, with each application configured
+under a unique key beneath the top-level `applications` key.
+
+For example, it is possible to deploy multiple runtimes in an application container - one PHP, one JavaScript and one
+Python - for the frontend of a deployed site.
```
Comment (lolautruche) [inline sites/friday/src/create-apps/app-reference/composable-image.md:46]:
```suggestion
It is possible to use multiple runtimes in a single application container, e.g. PHP, NodeJS and Python, while remaining in control of their versions.
```

File: sites/friday/src/create-apps/app-reference/composable-image.md [hunk 2]
```diff
@@ -0,0 +1,1352 @@
...
+
+In this case, the unified `{{< vendor/configfile "app" >}}` file would look like:
```
Comment (lolautruche) [inline sites/friday/src/create-apps/app-reference/composable-image.md:48]:
```suggestion
For example, the unified `{{< vendor/configfile "app" >}}` file may look like the following:
```

File: sites/friday/src/create-apps/app-reference/composable-image.md [hunk 3]
```diff
@@ -0,0 +1,1352 @@
...
+To override any part of a property, you have to provide the entire property.
+
+| Name               | Type                                                | Required | Set in instance? | Description                                                                                                                                                                                                                                                      |
+|--------------------|-----------------------------------------------------|----------|------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `stack`            | An array of [NixOs runtimes](#stack)                | Yes      | No               | The list of runtime images to use with a specific app. Format: <br>`<runtime>@<version>`.                                                                                                                                                                        |
```
Comment (lolautruche) [inline sites/friday/src/create-apps/app-reference/composable-image.md:73]:
```suggestion
| `stack`            | An array of [Nix packages](#stack)                | Yes      | No               | A list of packages from our collection of supported runtimes and/or from NixPkgs.                                                                                                                                                                        |
```

File: sites/friday/src/create-apps/app-reference/composable-image.md [hunk 4]
```diff
@@ -0,0 +1,1352 @@
...
+to the above.
+
+To specify another directory, for example for a [multi-app project](../multi-app/_index.md),
+use the [`source.root` property](#source).
+
+## Stack
+
+The ``stack`` keyword is used for defining runtimes and other binaries (like `wkhtmltopdf`, or `imagemagick`, see list
+below),
+as a YAML array, that you would like to install in your application container.
```
Comment (lolautruche) [inline sites/friday/src/create-apps/app-reference/composable-image.md:122]:
```suggestion
The ``stack`` keyword is used to define the list of all the packages your application needs to work, 
including language runtimes (e.g. `php`, `nodejs`, `python`...) and other utilities (e.g. `imagemagick`, `graphviz`, `yarn`...).
This list is defined as a YAML array.
```

File: sites/friday/src/create-apps/app-reference/composable-image.md [hunk 5]
```diff
@@ -0,0 +1,1352 @@
...
+The version is the major (`X`) and sometimes minor (`X.Y`) version numbers,
+depending on the service, as in the following table.
+Security and other patches are taken care of for you automatically.
```
Comment (lolautruche) [inline sites/friday/src/create-apps/app-reference/composable-image.md:125]: This is only true for the runtimes.
As for the other packages, it's up to the Nix maintainers.

File: sites/friday/src/create-apps/app-reference/composable-image.md [hunk 6]
```diff
@@ -0,0 +1,1352 @@
...
+Available tools and their supported versions:
+
+| **Tool**                                                             | **`runtime`** | **Supported `version`** |
+|----------------------------------------------------------------------|---------------|-------------------------|
+| [Face detector](https://www.thregr.org/~wavexx/software/facedetect/) | `facedetect`  | none                    |
+| [GNU Compiler Collection](https://gcc.gnu.org/)                      | `gcc`         | none                    |
+| [Graph visualization tools](https://graphviz.org/)                   | `graphviz`    | none                    |
+| [ImageMagick](http://www.imagemagick.org/)                           | `imagemagick` | none                    |
+| [WkHtmlToPdf](https://wkhtmltopdf.org/)                              | `wkhtmltopdf` | none                    |
+| [Yarn](https://classic.yarnpkg.com/)                                 | `yarn`        | none                    |
```
Comment (lolautruche) [inline sites/friday/src/create-apps/app-reference/composable-image.md:160]: I'm not sure we need to specify this.

File: sites/friday/src/create-apps/app-reference/composable-image.md [hunk 7]
```diff
@@ -0,0 +1,1352 @@
...
+    source:
+      root: "/"
+    stack:
+      - "php@{{% latest "php" %}}":
+        extensions:
+          - zip
+          - imagick
+```
+
+{{% /note %}}
```
Comment (lolautruche) [inline sites/friday/src/create-apps/app-reference/composable-image.md:281]: This should probably go to the PHP page

File: sites/friday/src/create-apps/app-reference/composable-image.md [hunk 8]
```diff
@@ -0,0 +1,1352 @@
...
+
+```yaml {configFile="app"}
+applications:
+  myapp:
+    stack:
+      - "python@3.12"
+      - "python312Packages.yq" # python package specific
+```
+
+{{< /codetabs >}}
```
Comment (lolautruche) [inline sites/friday/src/create-apps/app-reference/composable-image.md:299]: This should probably go to the Python page

File: sites/friday/src/create-apps/app-reference/composable-image.md [hunk 9]
```diff
@@ -0,0 +1,1352 @@
...
+
+Alternatively, if you need to include configuration options for your extensions, use either your ``php.ini`` file or [environment variables](/development/variables/set-variables.md):
```
Comment (lolautruche) [inline sites/friday/src/create-apps/app-reference/composable-image.md:301]: Should also go to the PHP page

**Review by lolautruche — CHANGES_REQUESTED** [review]

**Comment by chadwcarlson** [comment]
Questions from review:

- [`socket_family` default](https://composable-image-o3lpfiy-ucq44jg6ofare.eu-5.platformsh.site/create-apps/app-reference/composable-image.html#upstream) - Currently reads "Defaults to tcp for all [image types](https://composable-image-o3lpfiy-ucq44jg6ofare.eu-5.platformsh.site/create-apps/app-reference/composable-image.html#types) except PHP; for PHP image types the default is unix." Does the default value change at all for different stacks, or for the composable image does this just always default to `tcp`?
- [`protocol` default](https://composable-image-o3lpfiy-ucq44jg6ofare.eu-5.platformsh.site/create-apps/app-reference/composable-image.html#upstream) - Same here, currently reads "Default varies based on [image type](https://composable-image-o3lpfiy-ucq44jg6ofare.eu-5.platformsh.site/create-apps/app-reference/composable-image.html#types)." Is this true when `stack` is used? Is there a consistent default in the case of `stack`?

File: sites/platform/src/languages/php/_index.md [hunk 10]
```diff
@@ -598,3 +606,48 @@ See [complete working examples for C and Rust](https://github.com/platformsh-exa
...
+
+{{% note title="TODO" %}}
+@Jerome: is all of this still valid? `request_terminate_timeout`, `sizing_hints`, `xdebug`?
+{{% /note %}}
+
+Here is an example configuration:
+
+```yaml {configFile="app"}
+applications:
+  frontend:
```
Comment (chadwcarlson) [inline sites/platform/src/languages/php/_index.md:636]: This is Upsun, not Platform.sh config here.
